cardano-testnet/src/Testnet/Components/Query.hs

```diff
@@ -98,25 +106,111 @@ waitForEpochs epochStateView@EpochStateView{nodeConfigPath, socketPath} interval
   currentEpoch <- getCurrentEpochNo epochStateView
   waitUntilEpoch nodeConfigPath socketPath $ addEpochInterval currentEpoch interval
 
+
+-- | Wait for the requested number of blocks
+waitForBlocks
+  :: HasCallStack
+  => MonadIO m
+  => MonadTest m
+  => EpochStateView
+  -> Word64 -- ^ Number of blocks to wait
+  -> m BlockNo -- ^ The block number reached
+waitForBlocks epochStateView numberOfBlocks = withFrozenCallStack $ do
+  fmap BlockNo . flip evalStateT Nothing $ do
+    watchEpochStateUpdate epochStateView $ \(_, _, BlockNo blockNumber) -> do
+      get >>= \case
+        Nothing -> do
+          -- register starting block
+          Nothing <$ put (Just blockNumber)
+        Just startingBlockNumber ->
+          if blockNumber >= startingBlockNumber + numberOfBlocks
+           then pure (Just blockNumber)
+           else pure Nothing
```

Comment:
```suggestion
        Just startingBlockNumber -> pure $
          if blockNumber >= startingBlockNumber + numberOfBlocks
          then Just blockNumber
          else Nothing
```
You can put the `pure` outside of the `if` since it is in both branches